python3Packages.langchain-openai: init at 0.1.17

drupol opened the pull request:
Initially, this PR was just supposed to add `langchain-openai`... Turns out that I added it, but also some other of its friends, and also fix a couple of discrepancies along the way.

Therefore, this PR fix also #328033

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

drupol committed:
python3Packages.langchain-core: 0.2.9 -> 0.2.21

drupol committed:
python3Packages.langchain-openai: init at 0.1.17

drupol commented:
@ofborg build python3Packages.langchain-core python3Packages.langchain-openai

drupol commented:
@ofborg build python311Packages.langchain-core python311Packages.langchain-openai

drupol committed:
python3Packages.langchain: fix revision and hash

drupol committed:
python3Packages.langchain-community: 0.2.9 -> 0.2.7

sarahec commented:
Changes that still need to happen:

1. Review the files where you updated the hashes and make sure you also updated the checkout refs. (They should only be `langchain-core==` for the langchain-core file. Otherwise, use the name of the thing you're building.)
2. You're going to run into `ModuleNotFound` on [langchain-standard-tests](https://github.com/langchain-ai/langchain/tree/master/libs/standard-tests). I'm currently banging my head on that -- do we add `libs/standard-tests` to the module search path? Do we build it as a subpackage (but not put into the nix cache)?

drupol commented:
>     1. Review the files where you updated the hashes and make sure you also updated the checkout refs. (They should only be `langchain-core==` for the langchain-core file. Otherwise, use the name of the thing you're building.)

I think this is what I did and fixed.

>     2. You're going to run into `ModuleNotFound` on [langchain-standard-tests](https://github.com/langchain-ai/langchain/tree/master/libs/standard-tests). I'm currently banging my head on that -- do we add `libs/standard-tests` to the module search path? Do we build it as a subpackage (but not put into the nix cache)?

I fixed that too in a `preConfigure` flag.

sarahec commented:
> I fixed that too in a `preConfigure` hook.

Clever. TIL.

P.S. Closes #328033

sarahec commented:
`langchain-community` is still checking out `langchain-core`.

Also, if you look at #328033 we also need to roll back a couple of version numbers in other files (and fix checkout paths).

drupol commented:
> `langchain-community` is still checking out `langchain-core`.

Yes, on purpose. There's no tag for `langchain-community`... https://github.com/langchain-ai/langchain/tags

drupol committed:
python3Packages.langchain-chroma: 0.2.9 -> 0.1.2

drupol commented:
> There is in the releases: https://github.com/langchain-ai/langchain/releases/tag/langchain-community%3D%3D0.2.7

Woops ! Nice one. I fixed it in this PR too.

sarahec commented:
Apologies for how many loops we're performing.

If you run `nix-build -A python312Packages.langchain-community` you'll see that the checkout fails and falls back on langchain==0.2.9.

I had to look back in the commit history to find the correct fix:

```nix
    rev = "langchain-community==${version}";
    hash = "sha256-r0YSJkYPcwjHyw1xST5Zrgg9USjN9GOsvhV97imSFCQ=";
```

There was a broad change a month ago that started checking out `refs/tags/langchain-community==...` and that was the start of the breakage.

I tested my suggestion and it builds and checks properly. (Now if I only could figure out suggesting changes in a PR...)

drupol commented:
> Apologies for how many loops we're performing.
> 
> If you run `nix-build -A python312Packages.langchain-community` you'll see that the checkout fails and falls back on langchain==0.2.9.
> 
> I had to look back in the commit history to find the correct fix:
> 
> ```nix
>     rev = "langchain-community==${version}";
>     hash = "sha256-r0YSJkYPcwjHyw1xST5Zrgg9USjN9GOsvhV97imSFCQ=";
> ```
> 
> There was a broad change a month ago that started checking out `refs/tags/langchain-community==...` and that was the start of the breakage.
> 
> I tested my suggestion and it builds and checks properly. (Now if I only could figure out suggesting changes in a PR...)

But isn't all of this fixed in https://github.com/NixOS/nixpkgs/pull/328207/commits/0d32e0b76ab441c0b797fc035950824cd068cc88 ?

drupol commented:
@ofborg build python3Packages.langchain-core python3Packages.langchain-openai python3Packages.langchain-community
@ofborg build python311Packages.langchain-core python311Packages.langchain-openai python311Packages.langchain-community

sarahec commented:
> But isn't all of this fixed in [0d32e0b](https://github.com/NixOS/nixpkgs/commit/0d32e0b76ab441c0b797fc035950824cd068cc88) ?

These do show up in the tags, so yes that should do it. I'm checking now by building each piece individually.

sarahec commented:
Result of `nixpkgs-review pr 328207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/langchain-openai/default.nix:
```diff
@@ -0,0 +1,103 @@
+  };
+
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
```
```suggestion
    description = "Integration package connecting OpenAI and LangChain";
```

SuperSandro2000 commented on pkgs/development/python-modules/langchain-openai/default.nix:
```diff
@@ -0,0 +1,103 @@
+  pytestFlagsArray = [ "tests/unit_tests" ];
+
+  disabledTests = [
+    # These tests have network access
```
```suggestion
    # These tests require network access
```

sarahec commented:
Error when testing on Darwin: Langchain-core fails testing at `test_queue_for_streaming_via_sync_call` and `test_same_event_loop` due to the test comparing execution time with magic values. Suggest skipping.
2.

drupol committed:
python3Packages.langchain-azure-dynamic-sessions: init at 0.1.0

sarahec commented:
Result of `nixpkgs-review pr 328207` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>28 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

drupol committed:
python3Packages.langchain-huggingface: init at 0.0.3

sarahec commented:
I'm running a final `nixpkgs-review` on Linux and Darwin. Once those post, I'll approve.

drupol committed:
python3Packages.langchain-mongodb: init at 0.1.6

drupol commented:
OK I just pushed some other changes, can you please review those as well ? I will stop now adding stuff :)

sarahec commented:
Result of `nixpkgs-review pr 328207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>36 packages built:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-azure-dynamic-sessions</li>
    <li>python311Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

drupol committed:
python3Packages.langchain-text-splitters: 0.2.9 -> 0.2.2

drupol commented:
I just fixed `langchain-text-splitters` too.

sarahec commented:
Result of `nixpkgs-review pr 328207` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-azure-dynamic-sessions</li>
    <li>python311Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.langchain-mongodb</li>
    <li>python311Packages.langchain-mongodb.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-mongodb</li>
    <li>python312Packages.langchain-mongodb.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

sarahec commented:
> I just fixed `langchain-text-splitters` too, so your ticket can be completely done.

You, my friend, are a rock star. Can you edit your description to match what you've done and add `Closes #328033`?

(Had to restart Darwin for some reason. It's nearly done with the review.)

sarahec reviewed (approved):
LGTM. All builds and checks pass. 

A heroic effort!

drupol commented:
> > I just fixed `langchain-text-splitters` too, so your ticket can be completely done.
> 
> You, my friend, are a rock star. Can you edit your description to match what you've done and add `Closes #328033`?
> 
> (Had to restart Darwin for some reason. It's nearly done with the review.)

Done. The related issue will be automatically closed when this PR will be merged. Hopefully asap, I need it at work!

sarahec commented:
Result of `nixpkgs-review pr 328207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-azure-dynamic-sessions</li>
    <li>python311Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.langchain-mongodb</li>
    <li>python311Packages.langchain-mongodb.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-mongodb</li>
    <li>python312Packages.langchain-mongodb.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

sarahec commented:
@SuperSandro2000 Your changes were made. Can you mark as resolved? (Thanks!)

drupol commented:
Planning to self merge this at lunchtime (UTC, 10:00a). Hurry for the reviews :)

GaetanLepage commented on pkgs/development/python-modules/langchain-core/default.nix:
```diff
@@ -89,6 +86,12 @@ buildPythonPackage rec {
     '';
   };
 
+  disabledTests = lib.optionals stdenv.isDarwin [
+    # Langchain-core the following tests due to the test comparing execution time with magic values.
+    "test_queue_for_streaming_via_sync_call"
+    "test_same_event_loop"
+  ];
+
```
I would have this snippet before `passthru` but it's not a big deal, so feel free to ignore.

drupol replied:
As you don't really mind, I can take care of that during the next PR for adding `langchain` stuff.

GaetanLepage reviewed (commented):
Diff LGTM otherwise !

GaetanLepage commented:
Result of `nixpkgs-review pr 328207` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-azure-dynamic-sessions</li>
    <li>python311Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.langchain-mongodb</li>
    <li>python311Packages.langchain-mongodb.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-mongodb</li>
    <li>python312Packages.langchain-mongodb.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

GaetanLepage commented:
Result of `nixpkgs-review pr 328207` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
  </ul>
</details>
<details>
  <summary>34 packages built:</summary>
  <ul>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-azure-dynamic-sessions</li>
    <li>python311Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.langchain-mongodb</li>
    <li>python311Packages.langchain-mongodb.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-mongodb</li>
    <li>python312Packages.langchain-mongodb.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

GaetanLepage commented:
Result of `nixpkgs-review pr 328207` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
  </ul>
</details>
<details>
  <summary>34 packages built:</summary>
  <ul>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-azure-dynamic-sessions</li>
    <li>python311Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-mongodb</li>
    <li>python311Packages.langchain-mongodb.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-mongodb</li>
    <li>python312Packages.langchain-mongodb.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

GaetanLepage commented:
Result of `nixpkgs-review pr 328207` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
  </ul>
</details>
<details>
  <summary>18 packages failed to build:</summary>
  <ul>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain-azure-dynamic-sessions</li>
    <li>python311Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python311Packages.langchain-chroma</li>
    <li>python311Packages.langchain-chroma.dist</li>
    <li>python311Packages.langchain-community</li>
    <li>python311Packages.langchain-community.dist</li>
    <li>python311Packages.langchain-core</li>
    <li>python311Packages.langchain-core.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.langchain-mongodb</li>
    <li>python311Packages.langchain-mongodb.dist</li>
    <li>python311Packages.langchain-openai</li>
    <li>python311Packages.langchain-openai.dist</li>
    <li>python311Packages.langchain-text-splitters</li>
    <li>python311Packages.langchain-text-splitters.dist</li>
    <li>python311Packages.langchain.dist</li>
  </ul>
</details>
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-mongodb</li>
    <li>python312Packages.langchain-mongodb.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/langchain-mongodb/default.nix:
```diff
@@ -0,0 +1,77 @@
+  };
+
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
```
```suggestion
    description = "Integration package connecting OpenAI and LangChain";
```

SuperSandro2000 commented on pkgs/development/python-modules/langchain-mongodb/default.nix:
```diff
@@ -0,0 +1,77 @@
+
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
+    homepage = "https://github.com/langchain-ai/langchain";
```
We should probably link the subpages here. Is it https://github.com/langchain-ai/langchain/tree/master/libs/partners/mongodb ?

Applies to the other packages as well

SuperSandro2000 commented on pkgs/development/python-modules/langchain-mongodb/default.nix:
```diff
@@ -0,0 +1,77 @@
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
+    homepage = "https://github.com/langchain-ai/langchain";
+    changelog = "https://github.com/langchain-ai/langchain/releases/tag/v${version}";
```
```suggestion
    changelog = "https://github.com/langchain-ai/langchain/releases/tag/langchain-mongodb==${version}";
```

Can you also check the other packages for similar things?

SuperSandro2000 commented on pkgs/development/python-modules/langchain-huggingface/default.nix:
```diff
@@ -0,0 +1,80 @@
+
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
+    homepage = "https://github.com/langchain-ai/langchain";
```
```suggestion
    homepage = "https://github.com/langchain-ai/langchain/tree/master/libs/partners/huggingface";
```

SuperSandro2000 commented on pkgs/development/python-modules/langchain-huggingface/default.nix:
```diff
@@ -0,0 +1,80 @@
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
+    homepage = "https://github.com/langchain-ai/langchain";
+    changelog = "https://github.com/langchain-ai/langchain/releases/tag/v${version}";
```
```suggestion
    changelog = "https://github.com/langchain-ai/langchain/releases/tag/langchain-huggingface==${version}";
```

SuperSandro2000 commented on pkgs/development/python-modules/langchain-azure-dynamic-sessions/default.nix:
```diff
@@ -0,0 +1,76 @@
+
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
+    homepage = "https://github.com/langchain-ai/langchain";
```
```suggestion
    homepage = "https://github.com/langchain-ai/langchain/tree/master/libs/partners/azure-dynamic-sessions";
```

SuperSandro2000 commented on pkgs/development/python-modules/langchain-azure-dynamic-sessions/default.nix:
```diff
@@ -0,0 +1,76 @@
+  meta = {
+    description = "An integration package connecting OpenAI and LangChain";
+    homepage = "https://github.com/langchain-ai/langchain";
+    changelog = "https://github.com/langchain-ai/langchain/releases/tag/v${version}";
```
```suggestion
    changelog = "https://github.com/langchain-ai/langchain/releases/tag/langchain-azure-dynamic-sessions==${version}";
```

drupol commented:
@ofborg build python3Packages.langchain-openai python3Packages.langchain-mongodb python3Packages.langchain-huggingface python3Packages.langchain-azure-dynamic-sessions python3Packages.langchain-text-splitters python3Packages.langchain-chroma python3Packages.langchain-community
@ofborg build python311Packages.langchain-openai python311Packages.langchain-mongodb python311Packages.langchain-huggingface python311Packages.langchain-azure-dynamic-sessions python311Packages.langchain-text-splitters python311Packages.langchain-chroma python311Packages.langchain-community

drupol commented:
@ofborg build python3Packages.langchain-openai python3Packages.langchain-mongodb python3Packages.langchain-huggingface python3Packages.langchain-azure-dynamic-sessions python3Packages.langchain-text-splitters python3Packages.langchain-chroma python3Packages.langchain-community python3Packages.langchain python3Packages.langchain-core
@ofborg build python311Packages.langchain-openai python311Packages.langchain-mongodb python311Packages.langchain-huggingface python311Packages.langchain-azure-dynamic-sessions python311Packages.langchain-text-splitters python311Packages.langchain-chroma python311Packages.langchain-community python311Packages.langchain python311Packages.langchain-core
